### PR TITLE
chore(deps): Update json-rpc-engine

### PR DESCRIPTION
**Pull request by brad-decker**
## **Description**
Updates our usage of json-rpc-engine to use `@metamask/json-rpc-engine` and changes one usage to use the .destroy method instead of looping through the now private _middleware array. 

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23532?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Ensure that all operations of the extension still work, especially dapp interactions

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by brad-decker** [comment]
@metamaskbot update-policies

File: app/scripts/lib/rpc-method-middleware/handlers/eth-accounts.js [hunk 1]
```diff
@@ -21,8 +21,8 @@ export default requestEthereumAccounts;
 
 /**
  *
- * @param {import('json-rpc-engine').JsonRpcRequest<unknown>} _req - The JSON-RPC request object.
- * @param {import('json-rpc-engine').JsonRpcResponse<true>} res - The JSON-RPC response object.
+ * @param {import('@metamask/json-rpc-engine').JsonRpcRequest<unknown>} _req - The JSON-RPC request object.
```
Comment (MajorLift) [inline app/scripts/lib/rpc-method-middleware/handlers/eth-accounts.js:24]:
```suggestion
 * @param {import('@metamask/json-rpc-engine').JsonRpcRequest} _req - The JSON-RPC request object.
```

If we want to be verbose, this is also a possibility:

```suggestion
 * @param {import('@metamask/json-rpc-engine').JsonRpcRequest<import('@metamask/utils').JsonRpcParams>} _req - The JSON-RPC request object.
```

File: app/scripts/lib/rpc-method-middleware/handlers/log-web3-shim-usage.js [hunk 2]
```diff
@@ -25,8 +25,8 @@ export default logWeb3ShimUsage;
  */
 
 /**
- * @param {import('json-rpc-engine').JsonRpcRequest<unknown>} req - The JSON-RPC request object.
- * @param {import('json-rpc-engine').JsonRpcResponse<true>} res - The JSON-RPC response object.
+ * @param {import('@metamask/json-rpc-engine').JsonRpcRequest<unknown>} req - The JSON-RPC request object.
```
Comment (MajorLift) [inline app/scripts/lib/rpc-method-middleware/handlers/log-web3-shim-usage.js:28]:
```suggestion
 * @param {import('@metamask/json-rpc-engine').JsonRpcRequest} req - The JSON-RPC request object.
```

File: app/scripts/lib/rpc-method-middleware/handlers/request-accounts.js [hunk 3]
```diff
@@ -47,8 +47,8 @@ const locks = new Set();
 
 /**
  *
- * @param {import('json-rpc-engine').JsonRpcRequest<unknown>} _req - The JSON-RPC request object.
- * @param {import('json-rpc-engine').JsonRpcResponse<true>} res - The JSON-RPC response object.
+ * @param {import('@metamask/json-rpc-engine').JsonRpcRequest<unknown>} _req - The JSON-RPC request object.
```
Comment (MajorLift) [inline app/scripts/lib/rpc-method-middleware/handlers/request-accounts.js:50]:
```suggestion
 * @param {import('@metamask/json-rpc-engine').JsonRpcRequest} _req - The JSON-RPC request object.
```

File: app/scripts/lib/rpc-method-middleware/handlers/send-metadata.js [hunk 4]
```diff
@@ -25,8 +25,8 @@ export default sendMetadata;
  */
 
 /**
- * @param {import('json-rpc-engine').JsonRpcRequest<unknown>} req - The JSON-RPC request object.
- * @param {import('json-rpc-engine').JsonRpcResponse<true>} res - The JSON-RPC response object.
+ * @param {import('@metamask/json-rpc-engine').JsonRpcRequest<unknown>} req - The JSON-RPC request object.
```
Comment (MajorLift) [inline app/scripts/lib/rpc-method-middleware/handlers/send-metadata.js:28]:
```suggestion
 * @param {import('@metamask/json-rpc-engine').JsonRpcRequest} req - The JSON-RPC request object.
```

**Review by MajorLift — COMMENTED** [review]
Looks good overall! Just pointing out a few instances where `unknown` is passed into `JsonRpcRequest`'s `Params` generic argument. 

This is no longer possible because `Params` is now constrained by `JsonRpcParams`. We can fix this simply by omitting `Params` since its default value is also `JsonRpcParams`.

File: app/scripts/lib/rpc-method-middleware/handlers/institutional/mmi-open-add-hardware-wallet.js [hunk 5]
```diff
@@ -16,8 +16,8 @@ export default mmiOpenAddHardwareWallet;
  */
 
 /**
- * @param {import('json-rpc-engine').JsonRpcRequest} req - The JSON-RPC request object.
- * @param {import('json-rpc-engine').JsonRpcResponse<true>} res - The JSON-RPC response object.
+ * @param {import('@metamask/json-rpc-engine').JsonRpcRequest} req - The JSON-RPC request object.
```
Comment (MajorLift) [inline app/scripts/lib/rpc-method-middleware/handlers/institutional/mmi-open-add-hardware-wallet.js:19]: For consistency:

```suggestion
 * @param {import('@metamask/json-rpc-engine').JsonRpcRequest<import('@metamask/utils').JsonRpcParams>} req - The JSON-RPC request object.
```

**Comment by brad-decker** [comment]
For some reason a unit test is showing a snapshot change in CI but not locally. @NiranjanaBinoy and/or @MajorLift could you pull this locally and run tests to see if you can reproduce the snapshot change?

**Comment by MajorLift** [comment]
@brad-decker All jest tests are passing on local for me. Not seeing the snapshot change. :(

**Comment by brad-decker** [comment]
@metamaskbot update-policies

**Pull request closed without merging**